### PR TITLE
OLH-1747 Allow our stub API to respond to PUT requests

### PR DESCRIPTION
**Pull request by aenciso**
### What changed

Added a new `mfa-methods/{mfaidentifier}` route to the template and a corresponding handler

### Why did it change

To allow our frontend to call the new method and get a response

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

## Testing

Unit tests. Deployed to dev and tested manually

## How to review

Locally: Run `sam local start-api` and send a put request
Dev: Send a put request to the new route

### CODE REVIEW

**Review by alex9smith — COMMENTED** [review]
Nice one, I like the new validation function. 

I think it would be clearer if you moved the handler out to a separate file. That'd match the structure we have in the OIDC stub where we have separate files for each handler. 
You could move the `formatResponse` and `validateFields` up to a new file in the `/common` folder as well so we can use them in the other lambdas

**Comment by aenciso** [comment]
> Nice one, I like the new validation function.
> 
> I think it would be clearer if you moved the handler out to a separate file. That'd match the structure we have in the OIDC stub where we have separate files for each handler. You could move the `formatResponse` and `validateFields` up to a new file in the `/common` folder as well so we can use them in the other lambdas

Sure, I'll refactor it.

File: src/method-management/method-management.ts [hunk 1]
```diff
@@ -39,24 +38,56 @@ export const createMfaMethodHandler = async (
...
+  try {
+    assert(event.body, "no body provided");
+    const body = JSON.parse(event.body);
+    const mfaIdentifier = event.pathParameters?.mfaIdentifier;
+    const {
+      email,
+      credential,
+      otp,
+      mfaMethod: {
+        priorityIdentifier = undefined,
```
Comment (peterfajemisincabinetoffice) [inline src/method-management/method-management.ts:67]: Hi Alfonso, should this nmot include the mfaIndetifier as well?

Comment (peterfajemisincabinetoffice) [reply src/method-management/method-management.ts:67]: See here:  https://github.com/govuk-one-login/openapi-specs/blob/3722748aa2ea81f05c68f899939031cbb147f6d7/auth/method-management-api.yaml#L168

Comment (peterfajemisincabinetoffice) [reply src/method-management/method-management.ts:67]: Thought it should accept something like this:

{
  "email": "testuser@testexampledomain.co.uk",
  "credential": "SMS",
  "otp": "123456",
  "mfaMethod": {
    "mfaIdentifier": 1213,
    "priorityIdentifier": "SECONDARY",
    "mfaMethodType": "SMS",
    "endPoint": "PHONE",
    "methodVerfied": true
  }
}

Comment (aenciso) [reply src/method-management/method-management.ts:67]: @peterfajemisincabinetoffice well spotted. I messaged the auth team and they agree there is a discrepancy and sending the identifier in the body is not needed. They are going to create a PR for it. Once is done and merged, I'll review this PR to make sure it's aligned with the open api spec.

**Review by peterfajemisincabinetoffice — APPROVED** [review]
Approving on the basis that any new update to the Auth API should be a seaprate work and we should not block ongoing work due to dependency from another team.

**Pull request merged**